runtime/jcl/common/thread.cpp

```diff
@@ -385,4 +397,114 @@ Java_java_lang_Thread_startImpl(JNIEnv *env, jobject rcv, jlong millis, jint nan
 	vmFuncs->internalExitVMToJNI(currentThread);
 }
 
+#if JAVA_SPEC_VERSION >= 19
+/* static native Thread currentCarrierThread(); */
+jobject JNICALL
+Java_java_lang_Thread_currentCarrierThread(JNIEnv *env, jclass clazz)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	jobject result = vmFuncs->j9jni_createLocalRef(env, currentThread->threadObject);
+	vmFuncs->internalExitVMToJNI(currentThread);
+
+	return result;
+}
+
+/* native void setCurrentThread(Thread thread); */
+void JNICALL
+Java_java_lang_Thread_setCurrentThread(JNIEnv *env, jclass clazz, jobject thread)
+{
+	Assert_JCL_unimplemented();
+}
+
+/* static native Object[] extentLocalCache(); */
+jobject JNICALL
+Java_java_lang_Thread_extentLocalCache(JNIEnv *env, jclass clazz)
+{
+	Assert_JCL_unimplemented();
+	return NULL;
+}
+
+/* static native void setExtentLocalCache(Object[] cache); */
+void JNICALL
+Java_java_lang_Thread_setExtentLocalCache(JNIEnv *env, jclass clazz, jobjectArray cache)
+{
+	Assert_JCL_unimplemented();
+}
+
+/* private static native StackTraceElement[][] dumpThreads(Thread[] threads); */
+jobjectArray JNICALL
+Java_java_lang_Thread_dumpThreads(JNIEnv *env, jclass clazz, jobjectArray threads)
+{
+	Assert_JCL_unimplemented();
+	return NULL;
+}
+
+/* private static native Thread[] getThreads(); */
+jobjectArray JNICALL
+Java_java_lang_Thread_getThreads(JNIEnv *env, jclass clazz)
+{
+	J9VMThread *currentThread = (J9VMThread*)env;
+	J9JavaVM *vm = currentThread->javaVM;
+	J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	jobjectArray result = NULL;
+	PORT_ACCESS_FROM_JAVAVM(vm);
+
+	vmFuncs->internalEnterVMFromJNI(currentThread);
+	vmFuncs->acquireExclusiveVMAccess(currentThread);
+
+	j9object_t *threads = (j9object_t*)j9mem_allocate_memory(sizeof(j9object_t) * vm->totalThreadCount, J9MEM_CATEGORY_VM_JCL);
+	if (NULL == threads) {
+		vmFuncs->setNativeOutOfMemoryError(currentThread, 0, 0);
+	} else {
+		j9object_t *currentThreadPtr = threads;
+		J9VMThread *targetThread = vm->mainThread;
+		UDATA threadCount = 0;
+
+		do {
+			j9object_t threadObject = targetThread->threadObject;
+
+			/* Only count live threads */
+			if (NULL != threadObject) {
+				if (J9VMJAVALANGTHREAD_STARTED(currentThread, threadObject) && (NULL != J9VMJAVALANGTHREAD_THREADREF(currentThread, threadObject))) {
+					*currentThreadPtr++ = threadObject;
+					threadCount += 1;
+				}
+			}
+			targetThread = targetThread->linkNext;
+		} while (targetThread != vm->mainThread);
+
+		J9Class *arrayClass = fetchArrayClass(currentThread, J9VMJAVALANGTHREAD_OR_NULL(vm));
+		if (NULL != arrayClass) {
+			j9object_t arrayObject = vm->memoryManagerFunctions->J9AllocateIndexableObject(currentThread, arrayClass, threadCount, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
```

Comment:
```suggestion
			j9object_t arrayObject = vm->memoryManagerFunctions->J9AllocateIndexableObject(currentThread, arrayClass, (U_32)threadCount, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
```

Reply:
Either that or change the type of `threadCount` and `i` (line 484) to `U_32`.